tools/test.iso8601.c

```diff
@@ -46,13 +48,9 @@ static struct crm_option long_options[] = {
 static void
 log_time_period(int log_level, crm_time_period_t * dtp, int flags)
 {
-    char *end = NULL;
-    char *start = NULL;
-
-    if(dtp) {
-        start = crm_time_as_string(dtp->start, flags);
-        end = crm_time_as_string(dtp->end, flags);
-    }
+    char *start = crm_time_as_string(dtp->start, flags);
+    char *end = crm_time_as_string(dtp->end, flags);
+    assert(start != NULL && end != NULL);
```

Comment:
We have a macro, CRM_ASSERT(), to use instead of the system assert()

Reply:
Thanks, force pushed.

Btw. that's really meant just as an defensive measure, since
`crm_time_as_string` is effectively a library symbol, defeating
most if not all static analyzers.

That `log_time_period` itself is no defended explicitly then
follows naturally -- it's used within the same compilation unit (which
is moreover sanely manageable for humans themselves if the file
is not excessively large).